File: android/src/main/java/app/independo/capacitorvoicerecorder/core/SegmentInfo.java
package app.independo.capacitorvoicerecorder.core;

/**
 * Information about a completed audio segment in segmented recording mode.
 *
 * @param sessionId Unique session identifier
 * @param index Zero-based index of this segment within the session
 * @param uri Full file path URI to the segment
 * @param fileName File name of the segment
 * @param msDuration Duration of this segment in milliseconds
 * @param mimeType MIME type of the segment file
 */
public record SegmentInfo(
    String sessionId,
    int index,
    String uri,
    String fileName,
    int msDuration,
    String mimeType
) {}

File: ios/Sources/VoiceRecorder/Core/SegmentInfo.swift
import Foundation

/// Information about a completed audio segment in segmented recording mode.
struct SegmentInfo {
    /// Unique session identifier.
    let sessionId: String
    /// Zero-based index of this segment within the session.
    let index: Int
    /// Full file path to the segment.
    let uri: String
    /// File name of the segment.
    let fileName: String
    /// Duration of this segment in milliseconds.
    let msDuration: Int
    /// MIME type of the segment file.
    let mimeType: String
}